src/core/qgsvectorlayer.h

```diff
@@ -401,9 +401,12 @@ class CORE_EXPORT QgsVectorLayer : public QgsMapLayer, public QgsExpressionConte
       /**
        * Constructor for LayerOptions.
        */
-      explicit LayerOptions( bool loadDefaultStyle = true, bool readExtentFromXml = false )
+      explicit LayerOptions( bool loadDefaultStyle = true,
```

Comment:
Can you deprecate this constructor, and make a new one which requires an explicit QgsCoordinateTransformContext value ? (We don't want ANY default constructed contexts)

Reply:
Yes, but what do we do with 
`explicit QgsVectorLayer( ... const QgsVectorLayer::LayerOptions &options = QgsVectorLayer::LayerOptions()` ?